vislib/src/sys/Log.cpp

```diff
@@ -97,96 +96,41 @@ void vislib::sys::Log::DebugOutputTarget::Msg(UINT level,
  * vislib::sys::Log::FileTarget::FileTarget
  */
 vislib::sys::Log::FileTarget::FileTarget(const char *path, UINT level)
-        : Target(level), stream(NULL) {
-
-    int newFile = -1;
-#ifdef _WIN32
-#if (_MSC_VER >= 1400)
-    if (_sopen_s(&newFile, path, _O_APPEND | _O_CREAT | _O_TEXT | _O_TRUNC 
-            | _O_WRONLY, _SH_DENYWR, _S_IREAD | _S_IWRITE) != 0) {
-        newFile = -1;
-    }
-#else /* (_MSC_VER >= 1400) */
-    newFile = _sopen(path, _O_APPEND | _O_CREAT | _O_TEXT | _O_TRUNC
-        | _O_WRONLY, _SH_DENYWR, _S_IREAD | _S_IWRITE);
-#endif /* (_MSC_VER >= 1400) */
-#else /* _WIN32 */
-    newFile = open(path, O_APPEND | O_CREAT | O_LARGEFILE | O_TRUNC
-        | O_WRONLY, S_IRWXU | S_IRWXG);
-#endif /* _WIN32 */
-    this->stream = (newFile == -1) ? NULL : 
-#ifdef _WIN32
-        _fdopen(newFile, "wc");
-#else /* _WIN32 */
-        fdopen(newFile, "w");
-#endif /* _WIN32 */
-
-    if (this->stream != NULL) {
-        this->filename = path;
-    } else {
-        this->filename.Clear();
-    }
+        : Target(level)/*, stream(NULL)*/ {
 
+    sink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(path, true);
+    sink->set_pattern(std_pattern);
+    logger = std::make_shared<spdlog::logger>("default_file", sink);
+    filename = path;
 }
 
 
 /*
  * vislib::sys::Log::FileTarget::FileTarget
  */
 vislib::sys::Log::FileTarget::FileTarget(const wchar_t *path, UINT level)
-        : Target(level), stream(NULL) {
-
-    int newFile = -1;
-#ifdef _WIN32
-#if (_MSC_VER >= 1400)
-    if (_wsopen_s(&newFile, path, _O_APPEND | _O_CREAT | _O_TEXT | _O_TRUNC
-            | _O_WRONLY, _SH_DENYWR, _S_IREAD | _S_IWRITE) != 0) {
-        newFile = -1;
-    }
-#else /* (_MSC_VER >= 1400) */
-    newFile = _wsopen(path, _O_APPEND | _O_CREAT | _O_TEXT | _O_TRUNC
-        | _O_WRONLY, _SH_DENYWR, _S_IREAD | _S_IWRITE);
-#endif /* (_MSC_VER >= 1400) */
-#else /* _WIN32 */
-    newFile = open(W2A(path), O_APPEND | O_CREAT | O_LARGEFILE | O_TRUNC
-        | O_WRONLY, S_IRWXU | S_IRWXG);
-#endif /* _WIN32 */
-
-    this->stream = (newFile == -1) ? NULL : 
-#ifdef _WIN32
-        _fdopen(newFile, "wc");
-#else /* _WIN32 */
-        fdopen(newFile, "w");
-#endif /* _WIN32 */
-
-    if (this->stream != NULL) {
-        this->filename = path;
-    } else {
-        this->filename.Clear();
-    }
-
+        : Target(level)/*, stream(NULL)*/ {
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> cvt;
```

Comment:
std::wstring_convert is deprecated in C++17.
codecvt_utf8<wchar_t> seems to be Windows-only (c.f. table on https://en.cppreference.com/w/cpp/locale/wstring_convert).